docs/DeveloperGuide.md

```diff
@@ -1,38 +1,462 @@
+- Pairing: `[CLIENT_FORMAT] : [PROPERTY_FORMAT]` 
+
+The text file of which Client, Property and Pairing is being stored is `client.txt`, `property.txt` and `pairing.txt` 
+respectively.
+
+![Add Client to Storage Diagram](diagrams/StorageAddClientSequenceDiagram.jpg)
+![Add Property to Storage Diagram](diagrams/StorageAddPropertySequenceDiagram.jpg)
+![Add Pair to Storage Diagram](diagrams/StorageAddPairSequenceDiagram.jpg)
+
+The three sequence diagram above shows the sequence of which the append operation is being invoked. All three
+operations are similar in operations but are invoked with different `parameter` and `path`.
+
+### Update To File
+The update operation happens when entries in ClientList and PropertyList is being deleted and entries the hash map of 
+PairingList is being removed.
+
+The sequence diagram of `updateClient`, `updateProperty` and `updatePair` can be seen below:
+![Update Client Sequence Diagram](diagrams/StorageUpdateClientSD.jpg)
+![Update Property Sequence Diagram](diagrams/StorageUpdatePropertySD.jpg)
+![Update Pairing Sequence Diagram](diagrams/StorageUpdatePairSD.jpg)
+
+Note that when delete operation is being invoked on client and property, the `updatePair` method will also be invoked to
+prevent entries retaining within pairingList after it has been deleted from clientList or propertyList.
+
+
+### List feature
+
+There are 3 main steps whenever a list command needs to be executed
+* When the user enters any command, it first needs be understood. That is handled by the ParseManager class.
+   Next, when the first word entered by the user is determined to be `list`, ParseManager itself then determines
+   the type of list command entered, including the tags. ParseListClient, ParseListProperty and ParseListEverything
+   are checkers to ensure that a valid command has been entered. ParseListClient and ParseListProperty also determine
+   if tags have been entered, and if those tags are valid.
+  ```
+        case COMMAND_LIST:
+            ArrayList<String> listCommandTypeAndFlags = getListCommandType(commandDetail);
+            boolean isListProperty = listCommandTypeAndFlags.get(0).trim().equals(PROPERTY_FLAG);
+            boolean isListClient = listCommandTypeAndFlags.get(0).equals(CLIENT_FLAG);
+            boolean isListEverything = listCommandTypeAndFlags.get(0).equals(EVERYTHING_FLAG);
+            if (isListProperty) {
+                return new ParseListProperty(listCommandTypeAndFlags.get(1));
+            } else if (isListClient) {
+                return new ParseListClient(listCommandTypeAndFlags.get(1));
+            } else if (isListEverything && listCommandTypeAndFlags.get(1).isEmpty()) {
+                return new ParseListEverything();
+            } else {
+                throw new UndefinedSubCommandTypeException(MESSAGE_INCORRECT_LIST_DETAILS);
+            }
+  ```
+This block of code is part of ParseManager. It determines the type of list operation(-client, 
```

Comment:
Maybe this could be better represented using a sequence diagram rather than showing the source code